test/node/bson_corpus_tests.js

```diff
@@ -48,40 +49,97 @@ function normalize(cEJ) {
   return JSON.stringify(JSON.parse(cEJ));
 }
 
-// tests from the corpus that we need to skip, and explanations why
+const parseErrorForDecimal128 = scenario => {
+  const skipRegex = /dqbsr|Inexact/;
+  for (const parseError of scenario.parseErrors) {
+    it(parseError.description, function () {
+      if (skipRegex.test(parseError.description)) {
+        this.skip();
+      }
+
+      expect(
+        () => BSON.Decimal128.fromString(parseError.string),
+        `Decimal.fromString('${parseError.string}') should throw`
+      ).to.throw(/not a valid Decimal128 string/);
+    });
+  }
+};
+
+const parseErrorForBinary = scenario => {
+  for (const parseError of scenario.parseErrors) {
+    it(parseError.description, () => {
+      // Currently the BSON Binary parseError tests only check parsing relating to UUID
+      // in the future this regex may need expansion
+      expect(() => EJSON.parse(parseError.string)).to.throw(/UUID/);
+    });
+  }
+};
+
+const parseErrorForRootDocument = scenario => {
+  for (const parseError of scenario.parseErrors) {
+    it(parseError.description, function () {
+      let caughtError;
+      try {
+        // Make sure not add anything more than this line to the try block
+        // Assertions, for example, throw and will mess up the checks below.
+        EJSON.parse(parseError.string);
+      } catch (error) {
+        caughtError = error;
+      }
+
+      if (/Null/.test(parseError.description)) {
+        expect(caughtError).to.be.instanceOf(Error);
+        expect(caughtError.message).to.match(/null bytes/);
+      } else if (/Bad/.test(parseError.description)) {
+        // There is a number of failing tests that start with 'Bad'
+        // so this check is essentially making the test optional for now
+        // This should assert that e is an Error and something about the message
+        expect(caughtError).to.satisfy(e => {
+          if (e instanceof Error) return true;
+          else this.skip();
+        });
+      } else {
+        expect(caughtError).to.be.instanceOf(Error);
+      }
+    });
+  }
+};
+
 const skipBSON = {
   'NaN with payload':
     'passing this would require building a custom type to store the NaN payload data.'
 };
-
+// tests from the corpus that we need to skip, and explanations why
 const skipExtendedJSON = {
   'Timestamp with high-order bit set on both seconds and increment':
     'Current BSON implementation of timestamp/long cannot hold these values - 1 too large.',
   'Timestamp with high-order bit set on both seconds and increment (not UINT32_MAX)':
     'Current BSON implementation of timestamp/long cannot hold these values - 1 too large.'
 };
 
+const SKIP_TESTS = new Map([
+  ...Object.entries(skipBSON),
+  ...Object.entries(skipExtendedJSON),
+  [
+    'All BSON types',
```

Comment:
can we create/link a ticket to unskip this?

Reply:
These were all skipped before this work I just reworked the logic that checked for the test name to make the runner more consistent, look for the `'All BSON types'` string on the deletions side. So I'm pretty sure there's already tickets for these if they aren't inherent limitations to our driver.